### ISSUE TITLE
Performance metrics for TF on AMD cards ?

### ISSUE THREAD

**Issue by boxerab** [ROCmSoftwarePlatform/tensorflow#12]
It would be useful to compare TF performance on AMD against equivalent NV cards,
for example 580 vs 1060.

**Comment by Mandrewoid** [comment]
If you can find or create some simple **small-ish benchmarks,** I know a few people with Nvidia cards that I can distribute them to.  I have a Vega FE that I will run them on as well. 
Part of the problem is finding a benchmark suite that automatically adequately represents the performance of your card. 
As is, benchmarks like Resnet, Vggnet run differently with different size batches.
What runs fastest on my Vega FE will overflow the Vram on my Gtx 650 ti for example

**Comment by xyang2013** [comment]
Hi @Mandrewoid, could you please let me know what driver you used for Vega FE (enterprise, non-enterprise, or open-sourced one)? Thank you

My system:
Vega FE
Ubuntu 16.04.4

**Comment by Mandrewoid** [comment]
https://rocm.github.io/ROCmInstall.html

**Comment by xyang2013** [comment]
@Mandrewoid Thank you

**Comment by boxerab** [comment]
Anecdotal : for simple CNN, RX470 4GB is faster than Tesla K80 for training.

**Comment by ws0zzg4569** [comment]
An interesting finding: The loss of my model implement with keras mix with tensorflow running on my RX480 not change durining training epoch, but just tried with GTX1070 with tf1.3 and keras, works fine. No idea what cause the issue.

**Comment by ws0zzg4569** [comment]
```
    sentence_input = Input(shape=(MAX_SENT_LENGTH,EmbeddingDim),dtype='float32')
    l_lstm = Bidirectional(LSTM(LSTM_UNIT))(sentence_input)
    sentEncoder = Model(sentence_input, l_lstm)

    full_Input = Input(shape=(MAX_SENTS,MAX_SENT_LENGTH,EmbeddingDim),dtype='float32')
    full_encoder = TimeDistributed(sentEncoder)(full_Input)
    l_lstm_sent = Bidirectional(LSTM(LSTM_UNIT))(full_encoder)

    preds = Dense(1, activation='sigmoid')(l_lstm_sent)

    model = Model(full_Input,preds)

    model.compile(loss='binary_crossentropy',
                  optimizer='rmsprop',
                  metrics=['acc','precision','recall','fmeasure'])

    model.summary()

    model.fit(x,y,
              validation_data=(val_x,val_y),
              epochs=EPOCH_NUM,
              batch_size=BATCH_SIZE,
              verbose=2)
```
Well, just a simple hierarchical LSTM, and interestingly, loss explosion since epoch 2, everything works fine on the GTX1070. The loss converges after several epochs. 

```
Epoch 1/30
 - 18s - loss: 0.7404 - acc: 0.4664 - precision: 0.2618 - recall: 0.2325 - fmeasure: 0.2170 - val_loss: 0.7087 - val_acc: 0.4718 - val_precision: 0.1935 - val_recall: 0.0905 - val_fmeasure: 0.0872
Epoch 2/30
 - 16s - loss: 7.2821 - acc: 0.0376 - precision: 0.9534 - recall: 1.9314 - fmeasure: 1.2599 - val_loss: 8.0998 - val_acc: 0.0000e+00 - val_precision: 1.0000 - val_recall: 2.1890 - val_fmeasure: 1.3485
Epoch 3/30
 - 16s - loss: 7.8855 - acc: 0.0000e+00 - precision: 1.0000 - recall: 2.0061 - fmeasure: 1.3306 - val_loss: 8.0998 - val_acc: 0.0000e+00 - val_precision: 1.0000 - val_recall: 2.1890 - val_fmeasure: 1.3485
Epoch 4/30
 - -28784s - loss: 7.8855 - acc: 0.0000e+00 - precision: 1.0000 - recall: 2.0544 - fmeasure: 1.3339 - val_loss: 8.0998 - val_acc: 0.0000e+00 - val_precision: 1.0000 - val_recall: 2.1890 - val_fmeasure: 1.3485
Epoch 5/30
 - 16s - loss: 7.8855 - acc: 0.0000e+00 - precision: 1.0000 - recall: 2.0705 - fmeasure: 1.3347 - val_loss: 8.0998 - val_acc: 0.0000e+00 - val_precision: 1.0000 - val_recall: 2.1890 - val_fmeasure: 1.3485
Epoch 6/30
 - 16s - loss: 7.8855 - acc: 0.0000e+00 - precision: 1.0000 - recall: 2.0196 - fmeasure: 1.3317 - val_loss: 8.0998 - val_acc: 0.0000e+00 - val_precision: 1.0000 - val_recall: 2.1890 - val_fmeasure: 1.3485
Epoch 7/30
 - 16s - loss: 7.8855 - acc: 0.0000e+00 - precision: 1.0000 - recall: 2.0394 - fmeasure: 1.3330 - val_loss: 8.0998 - val_acc: 0.0000e+00 - val_precision: 1.0000 - val_recall: 2.1890 - val_fmeasure: 1.3485
......
```
The out put should be a binary classifier, but according to the value of recall, there seems to be  strange values.

**Comment by boxerab** [comment]
Thanks. Are you able to put this in a jupyter notebook, with data? I could try to repro with a notebook.

**Comment by psteinb** [comment]
I am just in the works of doing such a comparison with this little benchmark [deeprace](https://github.com/psteinb/deeprace). I can release the code in the repo or a related repo with analysis code in due course. Drop me a thumbs up if you are interested.
